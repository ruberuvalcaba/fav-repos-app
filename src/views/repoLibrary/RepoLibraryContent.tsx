import React, { useContext, useEffect } from 'react'
import { ExternalLinkIcon, DeleteIcon } from '@chakra-ui/icons'
import { Tbody, Tr, Td, Link, Text } from '@chakra-ui/react'
import { RepoContext } from '../../store/ReposStore'
import { Repo } from '../../types'
import { formatDate } from '../../utils'
import { StarsCount, TextWithEllipsis } from '../sharedComponents'

const RepoLibraryContent = ({
  onRemoveRepo,
}: {
  onRemoveRepo: (id: string) => void
}): JSX.Element => {
  const { state, actions } = useContext(RepoContext)
  const { reposList } = state
  const hasData = reposList?.length > 0

  //Initial load of repos list
  useEffect(() => {
    const getData = async () => {
      await actions.getReposList()
    }
    getData()
  }, [])
  return (
    <Tbody>
      {hasData &&
        reposList.map((repo: Repo) => {
          return (
            <Tr key={repo.id}>
              <Td>
                <TextWithEllipsis>{repo.fullName}</TextWithEllipsis>
              </Td>
              <Td>
                <Text fontSize="xs">{repo.language}</Text>
              </Td>
              <Td>
                <Link href={repo.url} isExternal>
                  <TextWithEllipsis>
                    <ExternalLinkIcon mr={1} />
                    {repo.url}
                  </TextWithEllipsis>
                </Link>
              </Td>
              <Td>
                <Text fontSize="xs">{formatDate(repo.createdAt)}</Text>
              </Td>
              <Td isNumeric>
                <StarsCount stars={repo.stargazersCount} />
              </Td>
              <Td display="flex" justifyContent="center" cursor="pointer">
                <DeleteIcon
                  onClick={() => onRemoveRepo(repo.id)}
                  color="gray.400"
                  _hover={{
                    color: 'gray.600',
                  }}
                />
              </Td>
            </Tr>
          )
        })}
    </Tbody>
  )
}

export default RepoLibraryContent
