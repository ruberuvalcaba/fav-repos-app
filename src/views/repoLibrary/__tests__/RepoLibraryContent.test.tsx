import React from 'react'
import { Table } from '@chakra-ui/react'
import renderer from 'react-test-renderer'
import RepoLibraryContent from '../RepoLibraryContent'

it('RepoLibraryContent component renders correctly', () => {
  const component = renderer.create(
    <Table>
      <RepoLibraryContent onRemoveRepo={() => {}} />
    </Table>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
