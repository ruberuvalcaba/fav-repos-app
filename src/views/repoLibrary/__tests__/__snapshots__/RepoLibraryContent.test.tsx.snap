// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoLibraryContent component renders correctly 1`] = `
<table
  className="chakra-table css-0"
>
  <tbody
    className="css-0"
  />
</table>
`;
